### DISCUSSION TITLE
How to repost quoted to a post in a feed

### DISCUSSION THREAD

**Discussion by MatthiasThoemel** [MarshalX/atproto#408] [Questions] [answered]
Let's say: 

I have a post from a feed in form of a FeedViewPost element, which I got from the return value of get_author_feed(...) call. 

Now, I want to repost this post by quoting it and adding my comment to it. 

I do not find a "repost by quotes" or similar in the Client API.

In the web UI of Bluesky, there is the Repost button and than it asks for: publish or quoting.

How do I do a repost with quoting the original post and adding my comment to the repost?

**Comment by MarshalX** [comment]
Hello, 

We have the example here:
https://github.com/MarshalX/atproto/blob/a24ac55716de3f5749816ff99f292d7bc481090f/examples/advanced_usage/send_embed.py#L20-L24

**Reply by MatthiasThoemel** [reply]
Hi, 

the example does not do exactly what I want to do. I have already an original post which I want to quote in a new post. The object type of the original post is FeedViewPost. 

If I take the example source code, I tried to do this:

```
    # for test: take first found post
    originalPost = gFoundPosts[0]

    # create an embedded post Main object
    embed_post = models.AppBskyEmbedRecord.Main(record=models.create_strong_ref(originalPost))        
    
    # create the comment on the original post
    commentText = client_utils.TextBuilder().text('This is a wonderful post')

    # post the comment with quoted original post
    gClient.send_post(text=commentText, embed=embed_post)   
```
But I get an error on running this. Maybe because a FeedViewPost object is wrong here:

raise ModelError('Could not create strong ref from model')
atproto_client.exceptions.ModelError: Could not create strong ref from model

On the line with:

embed_post = models.AppBskyEmbedRecord.Main(record=models.create_strong_ref(originalPost)) 

How can I use the example for FeedViewPost objects

**Reply by MarshalX** [reply, marked as answer]
You have .cid and .uri fields in post view. This is everything that you need to create a strong ref object. You are passing the wrong type. `originalPost` is incorrect. You forgot to call .post

**Reply by MatthiasThoemel** [reply]
Thank you very much.
